packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java

```diff
@@ -513,6 +597,19 @@ public void readDirectoryAsync(String uriStr, Map<String, Object> options, Promi
           promise.reject("ERR_FILESYSTEM_CANNOT_READ_DIRECTORY",
             "Directory '" + uri + "' could not be read.");
         }
+      } else if (isSAFUri(getContext(), uri)) {
+        DocumentFile file = DocumentFile.fromTreeUri(getContext(), uri);
+        if (file == null || !file.exists() || !file.isDirectory()) {
+          promise.reject("ERR_FILESYSTEM_CANNOT_READ_DIRECTORY",
+            "Uri '" + uri + "' doesn't exist or isn't a directory.");
+          return;
+        }
+        DocumentFile[] children = file.listFiles();
+        List<String> result = new ArrayList<>();
+        for (DocumentFile child : children) {
+          result.add(child.getUri().toString());
```

Comment:
```suggestion
          result.add(child.getName());
```
looking at the rest of the implementation, it looks like this method should return an array of file names rather than URIs.

Reply:
That is the desired behavior here. You can't just add a file name to the SAF URI to get a valid SAF URI. This URI type is more complicated. I think we don't want to bother about joining base URI with the file name. That's why I've decided to return a list of uris.